dlt/destinations/athena/athena.py

```diff
@@ -309,15 +320,18 @@ def _get_table_update_sql(self, table_name: str, new_columns: Sequence[TColumnSc
 
         bucket = self.config.staging_config.bucket_url
         dataset = self.sql_client.dataset_name
+
         sql: List[str] = []
 
         # for the system tables we need to create empty iceberg tables to be able to run, DELETE and UPDATE queries
-        is_iceberg = self.schema.tables[table_name].get("write_disposition", None) == "skip"
+        # or if we are in iceberg mode, we create iceberg tables for all tables
+        is_iceberg = (self.schema.tables[table_name].get("write_disposition", None) == "skip") or (self._is_iceberg_table(self.schema.tables[table_name]) and not self.in_staging_mode)
```

Comment:
this "in staging mode" is still here unfortunately.. i don't know how to do it any other way.

Reply:
if you move the `get_load_table` to JobClient you can modify table format to keep iceberg (destination dataset) or not (staging dataset). OFC you need to pass is_staging flag to the method. I do not like it but probably it is a good compromise

Reply:
as discussed: use get_load_table() to
- adjust the precision
- set the table format
and pass it to the code below